cmd/gardenlet/app/migration.go

```diff
@@ -6,10 +6,150 @@ package app
 
 import (
 	"context"
+	"fmt"
+	"slices"
 
 	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardeneraccess "github.com/gardener/gardener/pkg/component/gardener/access"
+	shootsystem "github.com/gardener/gardener/pkg/component/shoot/system"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
 )
 
-func (g *garden) runMigrations(_ context.Context, _ logr.Logger) error {
+func (g *garden) runMigrations(ctx context.Context, log logr.Logger) error {
+	log.Info("Migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig")
+	if err := migrateAdminViewerKubeconfigClusterRoleBindings(ctx, log, g.mgr.GetClient()); err != nil {
+		return fmt.Errorf("failed migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig: %w", err)
+	}
+
 	return nil
 }
+
+// migrateAdminViewerKubeconfigClusterRoleBindings moves the ClusterRoleBindings granting access to the
+// shoot/adminkubeconfig and shoot/viewerkubeconfig subresources from the shoot-core-system managed resource to the
+// shoot-core-gardeneraccess managed resource.
+// TODO(vpnachev): Remove this after v1.132.0 has been released.
+func migrateAdminViewerKubeconfigClusterRoleBindings(ctx context.Context, log logr.Logger, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
```

Comment:
Cosmetic suggestion:

```suggestion
	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot}); err != nil {
```

[`client.MatchingLabels`](https://github.com/kubernetes-sigs/controller-runtime/blob/6e240927407d36a0440a51a89d3c8c84038d04ad/pkg/client/options.go#L607) is a type alias for `map[string]string` and can be use as such.